spec/builtin-set.test.sh

```diff
@@ -276,3 +276,68 @@ echo "$@"
 +
 
 ## END
+
+#### set -a exports variables
+set -a
+FOO=bar
+BAZ=qux
+printenv.py FOO BAZ
+## STDOUT:
+bar
+qux
+## END
+
+#### set +a stops exporting
+set -a
+FOO=exported
+set +a
+BAR=not_exported
+printenv.py FOO BAR
+## STDOUT:
+exported
+None
+## END
+
+#### set -o allexport (long form)
+set -o allexport
+VAR1=value1
+set +o allexport
+VAR2=value2
+printenv.py VAR1 VAR2
+## STDOUT:
+value1
+None
+## END
+
+#### variables set before set -a are not exported
+BEFORE=before_value
+set -a
+AFTER=after_value
+printenv.py BEFORE AFTER
+## STDOUT:
+None
+after_value
+## END
+
+#### set -a exports local variables
+set -a
+f() {
+  local ZZZ=zzz
+  printenv ZZZ
```

Comment:
Hm I was confused by how this works ... I didn't know that `/usr/bin/printenv` exists!

It's part of `coreutils` apparently

It is basically the same as our tool, except in the case of unexported vars


```
$ printenv ZZ YY bad
zz
yy
$ echo $?
1

$ spec/bin/printenv.py ZZ YY bad
zz
yy
None
```

I think we should use one or the other consistently

So far we use printenv.py everywhere